Precise threat model, mitigation, add sources (#4)
t-bast/lightning-docs: spam-prevention.md

## Mitigation strategies available today

It is not possible today to fully prevent this type of attacks, but we can make the attacker's job
harder by properly configuring channels:

`htlc_minimum_msat * max_accepted_htlcs`

As this equation is function of channel policy parameters, properly configuring them can help 

> partially mitigate them:

Reviewer: I would remove these lines, and instead update the sentence below.

Author: Updated with sentence below.

> It is not possible today to fully prevent this type of attacks, but a stacking cost can be derived
> for an attacker to spam a channel with the following formula:
> 
> `htlc_minimum_msat * max_accepted_htlcs`
> 
> As this equation is function of channel policy parameters, properly configuring them can help 
> partially mitigate them:

Reviewer: This is a bit unclear, it took me some time to understand what you mean there.
What about:

```suggestion
It is not possible today to fully prevent this type of attacks, but we can make the attacker's job harder by properly configuring channels:

  * the attacker needs to lock at least `htlc_minimum_msat * max_accepted_htlcs` of his own funds to completely fill a channel, so you should use a reasonable value for `htlc_minimum_msat` (1 sat is **not** a reasonable value for channels with a big capacity; it may be ok for small channels though)
```

Reviewer: I think you only partially took that change, so it's a bit more confusing now :)

> * use a reasonable value for `htlc_minimum_msat` (1 sat is **not** a reasonable value for channels
>   with a big capacity; it may be ok for small channels though)

Reviewer: ```suggestion
* the attacker needs to lock at least `htlc_minimum_msat * max_accepted_htlcs` of his own funds to completely fill a channel, so you should use a reasonable value for `htlc_minimum_msat` (1 sat is **not** a reasonable value for channels with a big capacity; it may be ok for small channels though)
```

Author: Adopted this sentence.

* they are running modified (malicious) versions of LN node implementations
* they are able to quickly create many seemingly unrelated nodes
* they may already have long-lived channels (good reputation)

> * they might probe in-real time channel balances to adjust their spams

Reviewer: nit:

```suggestion
* they might probe in real-time channel balances to adjust their spams
```

There are important properties of Lightning that we must absolutely preserve:

* payer and payee's anonymity
* trustless payments
* minimal (reasonable) barrier to entry as routing node
* minimal overhead/cost for legitimate payments
* minimal overhead to declare public paths to the network

> * discriminate unreasonably between different class of payments (e.g hodl HTLCs)

Reviewer: Can you explain this one? I'm not sure if you mean we should penalize these types or payments or not.

Author: Reworded, lmk if good enough.

Reviewer: Looks good!

And we must avoid creating opportunities for attackers to:

* penalize an honest node's relationship with its own honest peers
* make routing nodes lose non-negligible funds

> * costlessly profit from any solution (e.g stealing upfront payments)

Reviewer: What about:

```suggestion
* steal money (even tiny amounts) from honest nodes
```

> * ease further discover position in payment path

Reviewer: nit:

```suggestion
* more easily discover their position in a payment path
```
